app/scripts/services/authorization.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, $cacheFactory, Logger, $interval, DataService){
+    function AuthorizationService() {
+      this.currentProject = null;
+      this.cachedRulesByProject = $cacheFactory('rulesCache', {
+        number: 10
+      });
+    }
+
+    // Transform data from:
+    // rules = {resources: ["jobs"], apiGroups: ["extensions"], verbs:["create","delete","get","list","update"]}
+    // into:
+    // normalizedRules = {"extensions/jobs": ["create","delete","get","list","update"]}
+    // Also while looping through the rules, check if user can create/update any resource on the 'Add to project' so the button will be displayed.
+    var normalizeRules = function(rules) {
+      var normalizedRules = {};
+      var canAddToProject = null;
+
+      // Check if resource name meets one of following conditions, since those resources can't be create/update via `Add to project` page:
+      //  - 'projectrequests'
+      //  - subresource taht contains '/', eg: 'builds/source', 'builds/logs', ...
+      //  - resource ending with 'review' string: 'localsubjectaccessreviews', 'selfsubjectrulesreviews', 'subjectaccessreviews' 
+      var checkResource = function(resource) {
+        if (resource === "projectrequests" || _.contains(resource, "/") || resource.endsWith("reviews")) {
+          return false;
+        } else {
+          return true;
+        }
+      };
+      _.each(rules, function(rule) {
```

Comment:
The motivation behind having all the logic (data normalization + test if user 'canAddToProject') to go through the received rules only once. On the other hand I can understand that it can get hard to read.